NetworkPkg/WifiConnectionManagerDxe/WifiConnectionMgrImpl.c

```diff
@@ -1489,8 +1489,26 @@ WifiMgrOnTimerTick (
   }
 
   Nic = (WIFI_MGR_DEVICE_DATA *)Context;
+  if (Nic->ConnectPendingNetwork == NULL) {
+    DEBUG ((DEBUG_INFO, "[WiFi Connection Manager] No profile for connection, no scan triggered!\n"));
+    gBS->CloseEvent (Event);
+    return;
+  }
+
+  if (StrLen (Nic->ConnectPendingNetwork->SSId) < SSID_MIN_LEN) {
+    DEBUG ((DEBUG_INFO, "[WiFi Connection Manager] Invalid SSID length for connection, no scan triggered!\n"));
+    gBS->CloseEvent (Event);
+    return;
+  }
+
   NET_CHECK_SIGNATURE (Nic, WIFI_MGR_DEVICE_DATA_SIGNATURE);
 
+  if ((Nic->ConnectState == WifiMgrConnectedToAp) || (Nic->ConnectState == WifiMgrConnectingToAp)) {
+    DEBUG ((DEBUG_INFO, "[WiFi Connection Manager] Already connecting to AP, no scan triggered!\n"));
+    gBS->CloseEvent (Event);
+    return;
+  }
+

```

Comment:
These early returns from the function make some of the code in the `switch` below unreachable, so it seems to me that the logic should be combined in some way.